Model/statemachine.py

```diff
@@ -997,44 +997,63 @@ def cancel_coin_selec_0(self, vault, needed_fee, feerate):
         return coins
 
     def cancel_coin_selec_1(self, vault, needed_fee, feerate):
-        """Select the combination that results in the smallest overpayment"""
+        """Select the combination of fee-bumping coins that results in the
+        smallest overpayment possible.
+
+        The UTxO pool is laid out with large coins covering up to the reserve
+        and smaller coins used for a finer grained coin selection to avoid
+        overpayments.
+        First try to find the number of Vb (large) coins needed to cover for
+        the most part of the fees, then fill the gap with Vm (small) coins.
+        """
         coins = []
-
-        best_combination = None
-        min_fee_added = None
-        max_paying_combination = None
-        max_fee_added = 0
-        allocated_coins = vault.allocated_coins()
-        for candidate in itertools.chain.from_iterable(
-            itertools.combinations(allocated_coins, r)
-            for r in range(1, len(allocated_coins) + 1)
-        ):
-            added_fees = sum(
-                [c.amount - P2WPKH_INPUT_SIZE * feerate for c in candidate]
+        allocated_coins = sorted(vault.allocated_coins(), key=lambda c: c.amount)
+        # All vb coins are always larger than vm coins
+        vm_coins, vb_coins = (
+            allocated_coins[: -self.vb_coins_count],
+            allocated_coins[-self.vb_coins_count :],
+        )
+        # All vb coins are created equal
```

Comment:
This is not true if the `feerate_reserve_per_vault` moves. I think using the `self.Vb` function is better

Reply:
> This is not true if the feerate_reserve_per_vault moves.

How so? All Vb coins are created at the same time for the same value. Hmm do you mean we could reallocate another Vb coin of a higher value to the vault?

In any case it's just an optimization i can get rid of this variable.

> I think using the self.Vb function is better

It would be worse: Vb() at the current block might not be equal to the Vb coins' value we were allocated

Reply:
Yeah, Vb changes in time, and Vb coins are created at different times, and our allocation mechanism has a tolerance around Vb, so a vault's reserve won't necessarily have all Vb coins the same size.

Perhaps we can use a tolerance here too. Vb coins are those with ```coin.amount >= 0.7*self.Vb(height)```

If the vault's reserve coins have fallen below that tolerance, they will be considered as Vm coins, which will still be picked up by the selection algorithm, just with poor runtime efficiency

Reply:
> Perhaps we can use a tolerance here too. Vb coins are those with coin.amount >= 0.7*self.Vb(height)
> If the vault's reserve coins have fallen below that tolerance, they will be considered as Vm coins, which will still be picked up by the selection algorithm, just with poor runtime efficiency

I don't need this variable really, i'll just get rid of it. Implicitly we would do something similar though: we consider to be Vb the top `self.n_vb_coins` coins of the sorted by amount list of coins.

Reply:
yeah sure, that makes sense :)